bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableTable.java

```diff
@@ -226,4 +228,15 @@ private boolean call() {
       }
     };
   }
+
+  @Override
+  public <R> void batchCallback(
+      List<? extends Row> actions, Object[] results, Batch.Callback<R> callback)
+      throws IOException, InterruptedException {
+    // This check is intentional because HBase-2.x does not perform param validation before return.
+    if (actions.isEmpty()) {
+      return;
```

Comment:
This looks fishy, does hbase not invoke the callback?

Reply:
In HBase-2.x, If the `actions` is empty then Hbase ignores Callback.